RecoTauTag/RecoTau/python/tools/runTauIdMVA.py

```diff
@@ -715,20 +720,26 @@ def runTauID(self):
                 'inner:RecoTauTag/TrainingFiles/data/DeepTauId/deepTau_2017v2p6_e6_inner.pb',
                 'outer:RecoTauTag/TrainingFiles/data/DeepTauId/deepTau_2017v2p6_e6_outer.pb',
             ]
-            setattr(self.process,_deepTauName+self.postfix,cms.EDProducer("DeepTauId",
-                electrons                = cms.InputTag('slimmedElectrons'),
-                muons                    = cms.InputTag('slimmedMuons'),
-                taus                     = cms.InputTag(self.originalTauName),
-                pfcands                  = cms.InputTag('packedPFCandidates'),
-                vertices                 = cms.InputTag('offlineSlimmedPrimaryVertices'),
-                rho                      = cms.InputTag('fixedGridRhoAll'),
-                graph_file               = cms.vstring(file_names),
-                mem_mapped               = cms.bool(False),
-                version                  = cms.uint32(self.getDeepTauVersion(file_names[0])[1]),
-                debug_level              = cms.int32(0),
-                disable_dxy_pca          = cms.bool(True),
-                is_online                = cms.bool(False)
-            ))
+            if not self.useSonic:
+                # run the direct inference if not on sonic
+                setattr(self.process,_deepTauName+self.postfix,cms.EDProducer("DeepTauId",
+                    electrons                = cms.InputTag('slimmedElectrons'),
+                    muons                    = cms.InputTag('slimmedMuons'),
+                    taus                     = cms.InputTag(self.originalTauName),
+                    pfcands                  = cms.InputTag('packedPFCandidates'),
+                    vertices                 = cms.InputTag('offlineSlimmedPrimaryVertices'),
+                    rho                      = cms.InputTag('fixedGridRhoAll'),
+                    graph_file               = cms.vstring(file_names),
+                    mem_mapped               = cms.bool(False),
+                    version                  = cms.uint32(self.getDeepTauVersion(file_names[0])[1]),
+                    debug_level              = cms.int32(0),
+                    disable_dxy_pca          = cms.bool(True),
+                    is_online                = cms.bool(False)
+                ))
+            else:
+                if self.debug: print("running deeptau with SONIC")
```

Comment:
should be removed for production

Reply:
removed

Reply:
the original config setup have a lot of such printouts after `self.debug`. Still better to remove it?

Reply:
ah, I didn't notice it was behind `self.debug`.